core/domain/state_domain.py

```diff
@@ -1545,24 +1546,28 @@ def _validate_item_selec_input(self, strict: bool = False) -> None:
                 selections.
             ValidationError. Answer choices are empty or duplicate.
         """
-        # Here we use cast because we are narrowing down the type from
-        # various allowed cust. arg types to 'int', and here we are sure
-        # that the type is always going to be int because 'ItemInputSelection'
-        # customization args always contains 'minAllowableSelectionCount' key
-        # with int type of values.
-        min_value = cast(
-            int,
-            self.customization_args['minAllowableSelectionCount'].value
-        )
-        # Here we use cast because we are narrowing down the type from
-        # various allowed cust. arg types to 'int', and here we are sure
-        # that the type is always going to be int because 'ItemInputSelection'
-        # customization args always contains 'maxAllowableSelectionCount' key
-        # with int type of values.
-        max_value = cast(
-            int,
-            self.customization_args['maxAllowableSelectionCount'].value
-        )
+        min_value = self.customization_args['minAllowableSelectionCount'].value
+        max_value = self.customization_args['maxAllowableSelectionCount'].value
+
+        if min_value is not None and max_value is not None:
+            # Here we use cast because we are narrowing down the type from
+            # various allowed cust. arg types to 'int', and here we are sure
+            # that the type is always going to be int because
+            # 'ItemInputSelection' customization args always contains
+            # 'maxAllowableSelectionCount' key
+            # with int type of values.
+            min_value = cast(int, min_value)
+            # Here we use cast because we are narrowing down the type from
+            # various allowed cust. arg types to 'int', and here we are sure
+            # that the type is always going to be int because
+            # 'ItemInputSelection' customization args always contains
+            # 'maxAllowableSelectionCount' key
+            # with int type of values.
+            max_value = cast(int, max_value)
+        else:
```

Comment:
It's simpler to inverse the logic:
```
if min_value is None or max_value is None:
  raise
min_value = ...
max_value = ...
```

The fewer nesting - the better.

Reply:
Done